torch/csrc/autograd/python_variable.cpp

```diff
@@ -609,6 +609,7 @@ static int THPVariable_clear(THPVariable* self) {
     }
   }
   TORCH_INTERNAL_ASSERT(!isResurrectable((THPVariable*)self));
+  pybind11::gil_scoped_release no_gil;
```

Comment:
```suggestion
  {
    pybind11::gil_scoped_release no_gil;
    self->cdata = MaybeOwned<Variable>();
  }
```

Could you explicitly scope this around the destructor to ensure we don't add anything below by mistake in the future?

Reply:
That's a really good point.